src/mod/pthread_spinlock.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2025 Huawei Technologies Co., Ltd.
+ * SPDX-License-Identifier: 0BSD
+ */
+#include <assert.h>
+#include <pthread.h>
+
+#include <dice/chains/intercept.h>
+#include <dice/events/pthread.h>
+#include <dice/interpose.h>
+#include <dice/pubsub.h>
+
+#if defined(HAVE_SPINLOCK) && !defined(__OHOS__)
```

Comment:
can we rename this as `HAVE_PTHREAD_SPINLOCK` ?

Reply:
Actually, this check should not be necessary. We should simply not compile this file on when spinlock is not available.

The condition should be added to src/mod/CMakeLists.txt